test/unit/logger_json_google_test.exs

```diff
@@ -313,6 +313,76 @@ defmodule LoggerJSONGoogleTest do
     assert log["error"]["initial_call"] == "Elixir.Foo.bar/3"
   end
 
+  describe "google_error_reporter" do
+    setup do
+      Logger.configure_backend(LoggerJSON, metadata: :all)
+      :ok = Application.put_env(:logger_json, :google_error_reporter, %{})
+      on_exit(fn -> :ok = Application.delete_env(:logger_json, :google_error_reporter) end)
+    end
+
+    test "formats elixir error for google" do
+      stacktrace = [
+        {Foo, :bar, 0, [file: 'foo/bar.ex', line: 123]},
+        {Foo.Bar, :baz, 1, [file: 'foo/bar/baz.ex', line: 456]}
+      ]
+      Logger.metadata(crash_reason: {%RuntimeError{message: "oops"}, stacktrace})
+
+      log =
+        capture_log(fn -> Logger.error("hello") end)
+        |> Jason.decode!()
+
+      assert log["message"] ==
+               """
+               Elixir.RuntimeError: oops
+               \tfoo/bar.ex:123:in `Elixir.Foo.bar/0'
+               \tfoo/bar/baz.ex:456:in `Elixir.Foo.Bar.baz/1'
+               """
+               |> String.trim_trailing()
+    end
+
+    test "formats erlang error for google" do
+      stacktrace = [
+        {Foo, :bar, [], []},
+        {Foo, :bar, 0, [file: 'foo/bar.ex', line: 123]},
+        {Foo.Bar, :baz, 1, [file: 'foo/bar/baz.ex', line: 456]}
+      ]
+      Logger.metadata(crash_reason: {:undef, stacktrace})
+
+      log =
+        capture_log(fn -> Logger.error("hello") end)
+        |> Jason.decode!()
+
+      assert log["message"] ==
+               """
+               Elixir.UndefinedFunctionError: function Foo.bar/0 is undefined (module Foo is not available)
+               \tfoo/bar.ex:123:in `Elixir.Foo.bar/0'
+               \tfoo/bar/baz.ex:456:in `Elixir.Foo.Bar.baz/1'
+               """
+               |> String.trim_trailing()
+    end
+
+    test "adds @type metadata" do
+      Logger.metadata(crash_reason: {%RuntimeError{message: "oops"}, []})
+
+      log =
+        capture_log(fn -> Logger.error("hello") end)
+        |> Jason.decode!()
+
+      assert log["@type"] == "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent"
+    end
+
+    test "adds google_error_reporter metadata" do
+      :ok = Application.put_env(:logger_json, :google_error_reporter, %{serviceContext: %{service: "myapp", version: "abc123"}})
```

Comment:
I would prefer to use a keyword list here to match elixir configuration conventions. However, doing so results in an error about not being able to JSON serialize keyword lists.

What's odd is that you can pass keywords as Logger metadata e.g. `Logger.error("message", service_context: [ service: "myapp", version: "abc123" ])` and that works fine.

So I'm not quite sure how to approach this. Anyway, it is desirable to allow for a configured `serviceContext` so that GER reports which service and app version generated the error.